tests/unit/setup-rendering-context-test.js

```diff
@@ -0,0 +1,140 @@
+import { module, test } from 'qunit';
+import Service from '@ember/service';
+import Component from '@ember/component';
+import { helper } from '@ember/component/helper';
+import {
+  setupContext,
+  setupRenderingContext,
+  teardownContext,
+} from 'ember-test-helpers';
+import hasEmberVersion from 'ember-test-helpers/has-ember-version';
+import hasjQuery from '../helpers/has-jquery';
+import { setResolverRegistry } from '../helpers/resolver';
+import hbs from 'htmlbars-inline-precompile';
+
+module('setupRenderingContext', function(hooks) {
+  if (!hasEmberVersion(2, 4)) {
+    return;
+  }
+
+  hooks.before(function() {
+    setResolverRegistry({
+      'service:foo': Service.extend({ isFoo: true }),
+      'template:components/template-only': hbs`template-only component here`,
+      'component:js-only': Component.extend({
+        classNames: ['js-only'],
+      }),
+      'helper:jax': helper(([name]) => `${name}-jax`),
+      'template:components/outer-comp': hbs`outer{{inner-comp}}outer`,
+      'template:components/inner-comp': hbs`inner`,
+    });
+  });
+
+  hooks.beforeEach(function() {
+    setupContext(this);
+    setupRenderingContext(this);
+  });
+
+  hooks.afterEach(function() {
+    teardownContext(this);
+  });
+
+  test('render exposes an `.element` property', function(assert) {
+    return this.render(hbs`<p>Hello!</p>`).then(() => {
+      assert.equal(this.element.textContent, 'Hello!');
+    });
+  });
```

Comment:
since we're using headless Chrome we should be able to use native async/await in the tests, right?

Reply:
Yes, we can, but we have to take care not to use it in non-test parts.  I think I have a nice eslint rule for this, will try it out...